name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TastytradePDFExtractor.java

```diff
@@ -0,0 +1,529 @@
+            //    UNSOLICITED
+            //    PRODUCT DESCRIPTION UNDER
+            //    SEPARATE COVER
+            //    CUSIP: 74347X831
+            section -> section.optional()
+            .attributes("type", "tradedate", "name", "shares", "price", "amount", "cusip")
+            .match("(?<type>(BOUGHT|SOLD))[\\s]+(?<tradedate>[\\d]{2}/[\\d]{2}/[\\d]{2})[\\s]+[A-Z][\\s]+"
+                            + "(?<name>[\\/\\& \\w]+)[\\s]+"
+                            + "(?<shares>[\\.,\\d]+)[\\s]+"
+                            + "(\\$)?(?<price>[\\.,\\d]+)[\\s]+"
+                            + "(\\$)?(?<amount>[\\.,\\d]+)[\\s]*$")
+            .match("^CUSIP:[\\s]+(?<cusip>[A-Z0-9]+)")
+            .assign((t, v) -> {
+                t.setNote("");
+                if (v.get("type").equals("SOLD"))
+                    t.setType(PortfolioTransaction.Type.SELL);
+                v.put("wkn", v.get("cusip"));
+                t.setCurrencyCode("USD");
+                v.put("currency", "USD");
+                t.setSecurity(getOrCreateSecurity(v));
+                t.setDate(asDate(v.get("tradedate")));
+                t.setShares(asShares(v.get("shares"), "en", "US"));
+                t.setAmount(asAmount(v.get("amount"), "en", "US"));
+                checkAndSetFee(calculateFee(v), t, type.getCurrentContext());
+            }))
+
+            .wrap(BuySellEntryItem::new));
+
+        // Expired Option
+        //    EXPIRED 04/06/23 M PUT  TQQQ   04/06/23    27 1  
+        //    PROSHARES ULTRAPRO QQQ
+        //    OPTION EXPIRATION - EXPIRED
+        //    Security Number: 9LVCJB6
+        // Option Assignment
+        //    ASG 06/09/23 C CALL UPRO   06/09/23    40 1  
+        //    PROSHARES ULTRAPRO S&P 500
+        //    A/E 9SMCXK2 1 ASSIGNED
+        //    Security Number: 9SMCXK2
+        Block optionExpiration = new Block("^(EXPIRED|ASG).*$");
+        type.addBlock(optionExpiration);
+        optionExpiration.set(new Transaction<BuySellEntry>().subject(() -> {
+            BuySellEntry entry = new BuySellEntry();
+            entry.setMonetaryAmount(Money.of("USD", 0));
+            return entry;
+        })        
+            .section("date", "type",  "otype", "underlaying", "expiration", "strike", "quantity", "description", "cusip")
+            .match("^(?<type>(EXPIRED|ASG))[\\s]+(?<date>[\\d]{2}/[\\d]{2}/[\\d]{2})[\\s]+[A-Z][\\s]+(?<otype>[PUTCAL]{3,4})[\\s]+"
+                            + "(?<underlaying>[/\\&\\w]+)[\\s]+"
+                            + "(?<expiration>[\\d]{2}/[\\d]{2}/[\\d]{2})[\\s]+"
+                            + "(?<strike>[,\\.\\d]+)[\\s]+"
+                            + "(?<quantity>[\\-]?[\\d]+)[\\s]*.*$")
+            .match("^(?<description>.*).*$")
+            .match("(^OPTION EXPIRATION.*$|^.*[\\s]+[A-Z0-9]+[\\s]+[\\d]+[\\s]+ASSIGNED.*)")
+            .match("Security Number:[\\s]+(?<cusip>[A-Z0-9]+)")
+            .assign((t, v) -> {
                v.put("currency", "USD");
```

Comment:
here is a problem...
![grafik](https://github.com/portfolio-performance/portfolio/assets/45203494/852931d3-b742-4139-9941-36f0f1836277)

Reply:
I've converted the file and changed the git configuration as mention on Stackoverflow.
How can I check the line endings?